modules/lib/lib-project/src/main/resources/lib/xp/project.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Functions to manage Content Projects.
+ *
+ * @example
+ * var projectLib = require('/lib/xp/project');
+ *
+ * @module project
+ */
+
+/**
+ * Creates a new Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id (alpha-numeric characters and hyphens allowed).
+ * @param {string} [params.displayName] Project's display name. Defaults to `params.name`.
+ * @param {string} [params.description] Project description.
+ * @param {string} [params.language] Default project language.
+ * @param {Object.<string, string[]>} [params.permissions] Project permissions. 1 to 5 properties where key is a role and value is an array of principals.
+ * @param {string} params.permissions.role - Role id (one of `owner`, `editor`, `author`, `contributor`, `viewer`)
+ * @param {string[]} params.permissions.principals - Array of principals
+ * @param {Object<string, boolean>} [params.readAccess] Read access settings.
+ * @param {boolean} params.readAccess.public Public read access (READ permissions for `system.everyone`).
+ *
+ * @returns {Object} Created project.
+ */
+exports.create = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.CreateProjectHandler');
+    bean.id = required(params, 'id');
+    bean.displayName = nullOrValue(params.displayName);
+    bean.description = nullOrValue(params.description);
+    bean.language = nullOrValue(params.language);
+    bean.permissions = __.toScriptValue(params.permissions);
+    bean.readAccess = __.toScriptValue(params.readAccess);
+    return __.toNativeObject(bean.execute());
+};
+
+/**
+ * Modifies an existing Content Project.
+ *
+ * @param {Object} params JSON with the parameters.
+ * @param {string} params.id Unique project id to identify the project.
+ * @param {string} [params.displayName] Project's display name.
+ * @param {string} [params.description] Project description.
+ * @param {string} [params.language] Default project language.
+
+ *
+ * @returns {Object} Modified project.
+ */
+exports.modify = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.project.ModifyProjectHandler');
+    bean.id = required(params, 'id');
+    bean.displayName = nullOrValue(params.displayName);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['nullOrValue' was used before it was defined.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5511138)